client/webserver/webserver.go

```diff
@@ -516,8 +516,23 @@ func (s *WebServer) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Can't listen on %s. web server quitting: %w", s.addr, err)
 	}
+
+	// Get the actual address that the listener is using.
+	tcpAddr, ok := listener.Addr().(*net.TCPAddr)
+	if !ok {
+		return nil, fmt.Errorf("listener.Addr() is not a *net.TCPAddr")
+	}
+
+	// If the IP is the IPv6 address [::], use the zero IPv4 address instead.
+	// This is a workaround for an issue where browers see [::] as an invalid IP
+	// when used in the connect-src directive of a CSP like we do below.
+	if tcpAddr.IP.Equal(net.IPv6unspecified) {
+		tcpAddr.IP = net.IPv4zero
```

Comment:
`tcpAddr` is a pointer.  Don't modify the listener's `net.TCPAddr` that it owns.
We have no reason to change the listener, we would just need to change the CSP line given to the browser.  However, the premise of the fix is misinformed as my other comment describes.